From sarah.a.sharp@linux.intel.com  Tue Jun  2 12:07:55 2009
From: Sarah Sharp <sarah.a.sharp@linux.intel.com>
Date: Thu, 14 May 2009 11:44:26 -0700
Subject: USB: xHCI: Fix interrupt moderation.
To: Greg KH <greg@kroah.com>
Message-ID: <20090514184426.GA5430@gamba.jf.intel.com>


Mask off the lower 16 bits of the interrupt control register, instead of
masking off the upper 16 bits.  The interrupt moderation interval field is
the lower 16 bytes, and is set to 0x4000 (1ms) by default.  The previous
code was adding 40 us to the default value, instead of setting it to 40
us.  This makes performance really bad.

Signed-off-by: Sarah Sharp <sarah.a.sharp@linux.intel.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 drivers/usb/host/xhci-hcd.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/usb/host/xhci-hcd.c
+++ b/drivers/usb/host/xhci-hcd.c
@@ -384,7 +384,7 @@ int xhci_run(struct usb_hcd *hcd)
 
 	xhci_dbg(xhci, "// Set the interrupt modulation register\n");
 	temp = xhci_readl(xhci, &xhci->ir_set->irq_control);
-	temp &= 0xffff;
+	temp &= ~ER_IRQ_INTERVAL_MASK;
 	temp |= (u32) 160;
 	xhci_writel(xhci, temp, &xhci->ir_set->irq_control);
 
